### DISCUSSION TITLE
[Match] Different passwords for Git artifacts based on the Match type

### DISCUSSION THREAD

**Discussion by XabierGoros** [fastlane/fastlane#18179] [Q&A]
I was wondering it it's possible to have different Git encryption passwords depending on the Match type. I think it's interesting to distinguish between developer certificate and provisioning profiles and the ones to be used by the CI on the production environment. Thus, developers won't be able to access to the production signing certificate since its password will never be compromised, and the development one will be enough and available for them.

Please, correct me if I'm asking a ridiculous question which doesn't have any sense.

**Comment by jskrepnek** [comment]
I think it's a good question.

I don't think this is possible with match within a single repository.  A simple work around is to host two code signing repositories, one for development, one for appstore/distribution/production.

**Reply by XabierGoros** [reply]
Yes @jskrepnek, that's the workaround I found fo far in case any other proper solution is suggested. As you said, there would be two lanes containing the following command, one for developers and other for the CI machine:

```ruby
match(
  type: $my_app_type, 
  app_identifier: $my_app_id,
  git_url: $my_git_url
)
```
The passphrase for the development repository could be shared without such security concerns comparing to the production one. Another initial thought I had was to keep the same repo with different branches like this:

```ruby
match(
  type: $my_app_type, 
  app_identifier: $my_app_id,
  git_url: $my_git_url,
  git_branch: $type_branch
)
```
However, I'm not sure if any developer could unintentionally override the production branch with the development environment.
